dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -1393,7 +1383,8 @@ RtpsUdpDataLink::RtpsReader::process_data_i(const RTPS::DataSubmessage& data,
       info.held_.insert(std::make_pair(seq, *sample));
       info.recvd_.insert(seq);
       link->deliver_held_data(id_, info, durable_);
-    } else if (info.recvd_.disjoint() || info.recvd_.cumulative_ack() != seq.previous()) {
+    } else if (info.recvd_.disjoint() ||
+               (!info.first_delivered_data_ && info.recvd_.cumulative_ack() != seq.previous())) {
```

Comment:
I don't think you want the check for info.first_delivered_data_ to be down this far. I'd put it back up alongside the check for recvd_.empty(). The rest of this logic should work correctly for both durable and non-durable if we've configured recvd_ correctly by the time the code gets here.

Reply:
OK, I will push a revised version.  Keep in mind that I'm working on the case where `process_heartbeat_i` has already inserted into `info.recvd_`.  The failed tests on GHPRB are passing on my local build with this update.